### PR TITLE
implement DNS SRV lookup

### PR DESCRIPTION
**Pull request by kjx98**
if server host is "", look DNS SRV og domain for host addr

### CODE REVIEW

**Comment by kjx98** [comment]
move DNS SRV lookup under NewClient func

File: _example/example.go [hunk 1]
```diff
@@ -5,18 +5,19 @@ import (
 	"crypto/tls"
 	"flag"
 	"fmt"
-	"github.com/mattn/go-xmpp"
+	"github.com/kjx98/go-xmpp"
```
Comment (mattn) [inline _example/example.go]: Please do not change this.

File: _example/example.go [hunk 2]
```diff
@@ -5,18 +5,19 @@ import (
...
 )
 
-var server = flag.String("server", "talk.google.com:443", "server")
+var server = flag.String("server", "", "server")
 var username = flag.String("username", "", "username")
 var password = flag.String("password", "", "password")
 var status = flag.String("status", "xa", "status")
 var statusMessage = flag.String("status-msg", "I for one welcome our new codebot overlords.", "status message")
-var notls = flag.Bool("notls", false, "No TLS")
+var notls = flag.Bool("notls", true, "No TLS")
```
Comment (mattn) [inline _example/example.go]: ditto

File: xmpp.go [hunk 3]
```diff
@@ -79,13 +79,7 @@ func containsIgnoreCase(s, substr string) bool {
...
 	addr := host
 
-	if strings.TrimSpace(host) == "" {
-		a := strings.SplitN(user, "@", 2)
-		if len(a) == 2 {
-			addr = a[1]
-		}
-	}
-	a := strings.SplitN(host, ":", 2)
+	a := strings.SplitN(addr, ":", 2)
```
Comment (mattn) [inline xmpp.go]: please do not change this.

Comment (mattn) [inline xmpp.go]: please do not change this since host contains user:pass@host:port

**Comment by mattn** [comment]
please remove Makefile.

**Comment by mattn** [comment]
If you ready to make PR, please reopen this.

**Comment by Neustradamus** [comment]
@mattn: Any news after more one year? Please look here:
- https://github.com/mattn/go-xmpp/pull/108

**Pull request closed without merging**